fix(Date): locale with plurals

jsomsanith opened the pull request:
**What is the problem this PR is trying to solve?**
Talend/ui contains a date-fns local plugged on our translation system.
But it doesn't support plurals. 

**What is the chosen solution to this problem?**
Support plurals

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

<!-- You can add more checkboxes here -->

**[ ] This PR introduces a breaking change**

<!-- if the PR introduces a breaking change, add the description here. So when you merge this PR, add this description into the [breaking change wiki](https://github.com/Talend/ui/wiki/BREAKING-CHANGE) in the next version -->

<!-- **Original Template** -->

<!-- https://github.com/Talend/tools/blob/master/tools-root-github/.github/PULL_REQUEST_TEMPLATE.md -->

annilsson commented on packages/components/src/DateFnsLocale/distanceInWords/distanceInWords.js:
```diff
@@ -0,0 +1,96 @@
+	return {
+		lessThanXSeconds: {
+			one: t('DATE_FNS_LESS_THAN_SECOND', { defaultValue: 'less than a second' }),
+			other: t('DATE_FNS_LESS_THAN_SECOND_plural', { defaultValue: 'less than [count] seconds' }),
```
Why not `{{count}}`? And should we not pass `count` to i18next, for special plural cases in other languages?

jsomsanith commented:
In fact no that's not an option :)
We already thought about that and using date-fns locales implies to `import` them. So if we want to support a new language, we have to change the code in the components, and integrate the new versions in all the apps.
This custom date-fns locale is a trick, to plug it to our translation system.

In this PR, I looked at date-fns locales code and implemented almost the same thing, replacing the strings with our translation calls `t()`

We can't use i18next plural system because we can't extract the plural keys as they are implicit. So I added explicitly the `*_plural` keys.

I used a replace to avoid creating all the translations at each format with a specific count. Here I create all the strings once, and just reuse them.

jsomsanith commented on packages/components/src/DateFnsLocale/distanceInWords/distanceInWords.js:
```diff
@@ -0,0 +1,96 @@
+		let result;
+		if (typeof distanceInWords[token] === 'string') {
+			result = distanceInWords[token];
+		} else if (count === 1) {
```
How can we manage key extraction with plurals if we let i18next manage plural formats ?

jare-talend replied:
plural is manual for the beginning

jare-talend replied:
For the moment, the extract string is not centralized. I think that the developpers when he adds a plural string, he need to add the plural version on the dictionnary.
